x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileDocument.java

```diff
@@ -74,8 +67,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             return builder;
         }
 
-        public Profile.ProfileUser toProfileUser(@Nullable String realmDomain) {
-            return new Profile.ProfileUser(username, roles, realm.getName(), realmDomain, email, fullName, displayName, active);
+        public Profile.ProfileUser toProfileUser(@Nullable String domainName) {
```

Comment:
`domainName` can come from `realm.getDomain().name()`, no? Is there a reason we're not using that?

Reply:
Short answer: We should just use the domain information from the ProfileDocument itself and I updated accordingly.

Long answer: Yes there was a reason. The domain recorded in the document is a snapshot of the configuration at the moment of creation. We agreed that the actual domain information will still be derived from the incoming authentication (if there is one). That is, the final domain info may not be the same as the one recorded in the doc. 
That said, the use cases so far in this PR do not see any difference in the two because the document is either newly created or just updated with the incomig authentication. So the two should always have the identical domain info. For simplicity, I agree that this method should not take parameter and just uses the doc's domain info.